packages/fxa-settings/src/components/Settings/ModalMfaProtected/index.tsx

```diff
@@ -0,0 +1,157 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import React from 'react';
+import { useForm } from 'react-hook-form';
+import Modal from '../Modal';
+import InputText from '../../InputText';
+import { useFtlMsgResolver } from '../../../models';
+import { FtlMsg } from 'fxa-react/lib/utils';
+import { EmailCodeImage } from '../../images';
+import Banner, { ResendCodeSuccessBanner } from '../../Banner';
+
+type ModalProps = {
+  email: string;
+  onSubmit: (code: string) => void;
+  onDismiss: () => void;
+  handleResendCode: () => void;
+  clearErrorTooltip: () => void;
+  localizedErrorBannerMessage?: string;
+  localizedErrorTooltipMessage?: string;
+  resendCodeLoading: boolean;
+  showResendSuccessBanner: boolean;
+};
+
+type FormData = {
+  confirmationCode: string;
+};
+
+export const ModalMfaProtected = ({
+  email,
+  onSubmit,
+  onDismiss,
+  handleResendCode,
+  clearErrorTooltip,
+  localizedErrorBannerMessage,
+  localizedErrorTooltipMessage,
+  resendCodeLoading,
+  showResendSuccessBanner,
+}: ModalProps) => {
+  const ftlMsgResolver = useFtlMsgResolver();
+
+  const { handleSubmit, register, formState } = useForm<FormData>({
+    mode: 'all',
+    defaultValues: {
+      confirmationCode: '',
+    },
+  });
+
+  const { isDirty, isValid } = formState;
+  const buttonDisabled = !isDirty || !isValid;
+
+  return (
+    <Modal
+      data-testid="modal-verify-session"
+      descId="modal-mfa-protected-desc"
+      headerId="modal-mfa-protected-title"
+      hasButtons={false}
+      onDismiss={onDismiss}
+    >
+      <form
+        onSubmit={handleSubmit(({ confirmationCode }) => {
+          onSubmit(confirmationCode.trim());
+        })}
+      >
+        <FtlMsg id="modal-mfa-protected-title">
+          <h2 id="modal-mfa-protected-title" className="font-bold text-xl">
+            Enter confirmation code
+          </h2>
+        </FtlMsg>
+        <FtlMsg id="modal-mfa-protected-subtitle">
+          <p className="text-base mt-1">to change your Mozilla account info</p>
+        </FtlMsg>
+        {showResendSuccessBanner && !localizedErrorBannerMessage && (
+          <ResendCodeSuccessBanner />
+        )}
+        {localizedErrorBannerMessage && (
+          <Banner
+            type="error"
+            content={{ localizedHeading: localizedErrorBannerMessage }}
+          />
+        )}
+
+        <EmailCodeImage />
+
+        <FtlMsg
+          id="modal-mfa-protected-instruction"
+          vars={{ email }}
+          elems={{
+            email: <span className="font-bold">{email}</span>,
+          }}
+        >
+          <p id="modal-mfa-protected-desc" className="my-6">
+            Enter the code that was sent to{' '}
+            <span className="font-bold">{email}</span> within 5 minutes. This
```

Comment:
We should make this configurable. Please file a follow up for this since it involves messing with the app's configuration.